packages/core-contracts/test/utils/Ownable.test.js

```diff
@@ -0,0 +1,196 @@
+const { ethers } = hre;
+const assert = require('assert/strict');
+
+const { ZERO_ADDRESS } = require('../test-libs/test-lib');
+
+describe('Ownable', () => {
+  describe('Constructor / Deployment', () => {
+    it('Should revert when no argument passed to Ctor', async () => {
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+
+      await assert.rejects(
+        async () => {
+          await OwnableParent.deploy();
+        },
+        {
+          name: 'Error',
+          message:
+            `missing argument:  in Contract constructor (count=0,` +
+            ` expectedCount=1, code=MISSING_ARGUMENT, version=contracts/5.4.1)`,
+        }
+      );
+    });
+
+    it('Should revert when owner parameter is passed the zero address', async () => {
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+
+      await assert.rejects(
+        async () => {
+          await OwnableParent.deploy(ZERO_ADDRESS);
+        },
+        {
+          name: 'Error',
+          message:
+            `VM Exception while processing transaction: reverted with` +
+            ` reason string 'Owner address cannot be 0x0'`,
+        }
+      );
+    });
+
+    it('Should set owner address on deployment', async () => {
+      const [owner] = await ethers.getSigners();
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+      const ownableParent = await OwnableParent.deploy(owner.address);
+      await ownableParent.deployed();
+
+      const contractOwner = await ownableParent.owner();
+
+      assert.equal(contractOwner, owner.address);
+    });
+
+    it('Should trigger OwnerChanged event on deployment', async () => {
+      const [owner] = await ethers.getSigners();
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+      const ownableParent = await OwnableParent.deploy(owner.address);
+      await ownableParent.deployed();
+
+      await new Promise((resolve) => {
+        ownableParent.once('OwnerChanged', (oldOwner, newOwner) => {
```

Comment:
We shouldn't check for emitted events like this.
See `packages/deployer/test/sample-project/test/OwnerModule.test.js` L58